File: backend/app/schemas/subject.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class SubjectBase(BaseModel):
    name: str
    description: Optional[str] = None

class SubjectCreate(SubjectBase):
    pass

class SubjectUpdate(SubjectBase):
    pass

class SubjectInDBBase(SubjectBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class Subject(SubjectInDBBase):
    pass

class TopicBase(BaseModel):
    subject_id: int
    name: str
    description: Optional[str] = None

class TopicCreate(TopicBase):
    pass

class TopicUpdate(TopicBase):
    pass

class TopicInDBBase(TopicBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class Topic(TopicInDBBase):
    pass

File: frontend/src/types/subject.ts
// Subject and Topic related types

export interface Subject {
  id: number;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
}

export interface Topic {
  id: number;
  subject_id: number;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
}